helm/cluster/templates/clusterapi/controlplane/kubeadmcontrolplane.yaml

```diff
@@ -58,5 +58,7 @@ spec:
     {{- $users | indent 4 }}
     {{- end }}
   replicas: {{ .Values.global.controlPlane.replicas }}
+  rolloutBefore:
```

Comment:
The usage of `Values.global` usually indicates customer is allowed to change it, is that something we really need or is that something only GS should do?

Reply:
Good question. This property controls how long before the control plane certificates expiration should the nodes be rolled (and certificates renewed).

I understand that some customers need to configure their own time periods to renew the certificates (and roll the nodes in this case).

In vintage the detault time before the certificate expiration when the rotation should be triggered is set to 180 days. But some customers need to rotate their certificates e.g. every 3 months.

Reply:
Do you know if the customers are able to control this in the vintage clusters?

Reply:
Ok thanks, the reason why I'm asking is there's a difference between `global` and `internal` 

`providerIntegration` is a static one and should never be changed
`internal` value represents only GS employees should change it
`global` is aiming customer values

Reply:
Apparently customers did not need to modify this configuration directly in the past, so we can make it internal.

Reply:
Should it go to `internal.advancedConfiguration.controlPlane`?

Reply:
> Apparently customers did not need to modify this configuration directly in the past

If customers do not need to modify this flag, then we don't even need a Helm value for it and it can be static. We can always make it configurable later and expose a Helm value.

Reply:
If we really (not maybe) know about some advanced use case where somebody from Giant Swarm has to tweak this config for some customer, then we can put it to `internal.advancedConfiguration.controlPlane.*`

Reply:
Basically, we want to avoid exposing Helm values if we don't have to do it. The smaller API (exposed Helm) the better.

Reply:
Thinking about it, I think it would be best to let customers decide.

You made a good point, it might better make this flexible. In case customer need to follow a policy which requires rolling certificates every 90 days or so it might be best to giving them the possibility

Reply:
I think we need to configure it in some cases (e.g. in case the customer needs a shorter renewal period), but it's OK if we set this property internally as GS.

I discussed it with Tobias from Team Phoenix and in his opinion it's better to make the property internal.

Reply:
I moved the `rolloutBefore` property to `internal.advancedConfiguration.controlPlane` and added validation,